parser-typechecker/src/Unison/TermParser.hs

```diff
@@ -489,12 +489,12 @@ namespaceBlock :: Var v => P v (BlockElement v)
 namespaceBlock = do
   _ <- reserved "namespace"
   -- need a version of verifyRelativeName that takes a `Token Name`
-  name <- verifyRelativeName importWordyId
+  name <- verifyRelativeVarName importWordyId
   let statement = (Binding <$> binding) <|> namespaceBlock
   _ <- openBlockWith "where"
   elems <- sepBy semi statement
   _ <- closeBlock
-  pure $ Namespace (Name.toString $ L.payload name) elems
+  pure $ Namespace (Text.unpack . Var.name $ L.payload name) elems
```

Comment:
What's the diff between `Var.name` and `Name.fromVar` again?

Reply:
`Var.name` takes a `Var v => v` to a `Text`. For vars of type `User`, that's just the `Text` contained within (plus a number if it's not 0).

Example:

```
Var.name (Symbol 0 (User "foo")) = "foo.bar"
Var.name (Symbol 1 (User "foo")) = "foo1"
```

And `Name.fromVar` just takes that output `Text` and calls `unsafeFromText` on it.